torchvision/io/image.py

```diff
@@ -10,7 +10,17 @@
 try:
     _load_library("image")
 except (ImportError, OSError) as e:
-    warn(f"Failed to load image Python extension: {e}")
+    msg = str(e)
+    if msg:
+        msg = f"Failed to load image Python extension: {msg}"
+    else:
+        msg = (
+            "Failed to load image Python extension. "
+            "If you don't plan on using image functionality from `torchvision.io`, you can ignore this warning. "
+            "Otherwise, there might be something wrong with your environment. "
+            "Did you have `libjpeg` or `libpng` installed before building `torchvision` from source?"
+        )
```

Comment:
isn't there always a `msg`? Even if there isn't, the text below seems always useful so I'd suggest to `yield from` or at least always show both `str(e)` and the new text

Reply:
> isn't there always a `msg`?

Nope. In case the file is just not there, 

https://github.com/pytorch/vision/blob/6a85ef24ba7a5d5879c4bd795972881ef5001c9b/torchvision/io/image.py#L11

just raises an `ImportError` with an empty string as message. See for example https://github.com/pytorch/vision/issues/7036#issuecomment-1406585192. 

> I'd suggest to `yield from`

What do you mean? We are turning an error into a warning here. Are you saying you want to raise an exception instead? In any case, `yield from` only works for generators, but not for chaining exceptions. Did you mean `raise ... from e`?

> always show both `str(e)` and the new text

Fair.

Reply:
Right sorry, we can't do `raise from` - just always show the new text :)